File: Zestawy Zadań/C++/4 - Tablice/Zestaw 2 zadania/4.2.6 - Tablica z L liczb rzeczywistych oraz srednia arytmetyczna + ile z nich wiekszych od sredniej.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int L, licznik=0;
    double * wiekszeodsredniej;
    double sum = 0, srednia = 0;
    cout << "Podaj liczbe z przedzialu (0 < L < 50): ";
    cin >> L;
    while (L < 0 || L > 50)
    {
        cout << "Podaj liczbe z przedzialu (0 < L < 50): ";
        cin >> L;
    }
    vector<double> liczby(L);
    for (int i = 0; i < L; i++)
    {
        cout << "Podaj liczbe: ";
        cin >> liczby[i];
        sum += liczby[i];
    }
    srednia = sum / L;
    for (int i = 0; i < L; i++)
    {
        if (srednia < liczby[i])
        {
            licznik++;
        }
    }
    cout << "Srednia arytmetyczna liczb: " << srednia << endl;
    cout << "Ilosc liczb wiekszych od sredniej: " << licznik << endl;
}

File: Zestawy Zadań/Python/4 - Tablice/Zestaw 2 zadania/4.2.6 - Tablica z L liczb rzeczywistych oraz srednia arytmetyczna + ile z nich wiekszych od sredniej.py
przedzial = int(input("Podaj liczbe z przedzialu (0 < L < 50): "))

licznik = 0
tab = [0] * przedzial

while przedzial < 0 or przedzial > 50:
    przedzial = int(input("Podaj liczbe z przedzialu (0 < L < 50): "))

for i in range(0, przedzial):
    tab[i] = float(input("Podaj liczbe do tabeli: "))

srednia = sum(tab) / przedzial

for i in range(0, przedzial):
    if srednia < tab[i]:
        licznik += 1

print("Średnia arytmetyczna liczb z tablicy: ", srednia)
print("Ilość liczb większych od sredniej: ", licznik)
